src/modules/roc_pipeline/config.h

```diff
@@ -82,12 +85,15 @@ struct SenderConfig {
     //! Channel mask.
     packet::channel_mask_t input_channels;
 
-    //! Number of samples for internal frames.
-    size_t internal_frame_size;
+    //! Duration of the internal frames, in nanoseconds.
+    core::nanoseconds_t internal_frame_length;
 
     //! Packet length, in nanoseconds.
     core::nanoseconds_t packet_length;
 
+    //! Size of the internal frames.
+    size_t internal_frame_size; 
```

Comment:
This looks a bit messy. Now we have `internal_frame_length` AND `internal_frame_size` and should keep them consistent.

Some components use `internal_frame_length` (Mixer), and some use `internal_frame_size` (ResamplerReader, ResamplerWriter).

This approach is error-prone. It will break if we will update one field but forget to update another. It will also break if we will have components in the pipeline with different channel masks.

I'd prefer to have only one field - `internal_frame_length` - and pass it to all components (including BackendDispatcher). Then every component will use `ns_to_size()`.

Thus I suggest to remove `internal_frame_size` (here and below in this file) and `DefaultInternalFrameSize`.

Reply:
done in most places. I need to test the code and go through it again because in a recent merge, it seems a few function names were changed. so I'm working on this part again.